Could not load js config file D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\strapi-server.js: Cannot find module '@strapi/utils/dist/file'

Error: Could not load js config file D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\strapi-server.js: Cannot find module '@strapi/utils/dist/file'   │    
│   Require stack:                                                                                                                                                                 │    
│   - D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\dist\server\services\image-optimizer-service.js                                                     │    
│   - D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\dist\server\services\index.js                                                                       │    
│   - D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\dist\server\index.js                                                                                │    
│   - D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\strapi-server.js                                                                                    │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\utils\dist\index.js                                                                                                       │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\commands\actions\admin\create-user\action.js                                                                  │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\commands\actions\admin\create-user\command.js                                                                 │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\commands\index.js                                                                                             │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\cli.js                                                                                                        │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\strapi\bin\strapi.js                                                                                                      │    
│       at loadJsFile (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\core\app-configuration\load-config-file.js:18:13)                                            │    
│       at Module.loadFile (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\core\app-configuration\load-config-file.js:37:14)                                       │    
│       at Object.loadPlugins (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\core\loaders\plugins\index.js:90:41)                                                 │    
│       at async Strapi.loadPlugins (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\Strapi.js:310:5)                                                               │    
│       at async Promise.all (index 3)                                                                                                                                             │    
│       at async Strapi.register (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\Strapi.js:340:5)                                                                  │    
│       at async Strapi.load (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\Strapi.js:424:5)                                                                      │    
│       at async Object.develop (D:\projects\strapi\my-project\node_modules\@strapi\admin\dist\_chunks\index-ffd2f664.js:1245:28)                                                  │    
│       at async develop (D:\projects\strapi\my-project\node_modules\@strapi\admin\dist\_chunks\develop-da585b1e.js:52:5)                                                          │    
│       at async Command.parseAsync (D:\projects\strapi\my-project\node_modules\commander\lib\command.js:923:5) 


**I can't use the plugin, can you tell me what it is?**

![image](https://github.com/marlokessler/strapi-plugin-image-optimizer/assets/82722153/1128cd60-0066-4e59-afe6-76bd737a43f2)

![image](https://github.com/marlokessler/strapi-plugin-image-optimizer/assets/82722153/8930cffb-2560-4360-894b-01227c5d7a31)

![image](https://github.com/marlokessler/strapi-plugin-image-optimizer/assets/82722153/ecd99f45-00f8-4870-9e93-2e66dec5bbbb)

![image](https://github.com/marlokessler/strapi-plugin-image-optimizer/assets/82722153/6a329eea-701f-40bb-ad5b-a18ec865ec20)

--- a/example/package.json
+++ b/example/package.json
@@ -11,11 +11,15 @@
   },
   "devDependencies": {},
   "dependencies": {
-    "@strapi/plugin-i18n": "4.11.7",
-    "@strapi/plugin-users-permissions": "4.11.7",
-    "@strapi/strapi": "4.11.7",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
+    "react-router-dom": "5.3.4",
+    "@strapi/plugin-i18n": "4.20.4",
+    "@strapi/plugin-users-permissions": "4.20.4",
+    "@strapi/strapi": "4.20.4",
     "better-sqlite3": "8.0.1",
-    "strapi-plugin-image-optimizer": "../"
+    "strapi-plugin-image-optimizer": "link:../",
+    "styled-components": "^5.2.1"
   },
   "author": {
     "name": "A Strapi developer"
@@ -24,7 +28,7 @@
     "uuid": "33f536a8-ba40-4664-bfb2-d50d6ab5abc8"
   },
   "engines": {
-    "node": ">=14.19.1 <=18.x.x",
+    "node": ">=18.0.0 <=20.x.x",
     "npm": ">=6.0.0"
   },
   "license": "MIT"

--- a/example/src/extensions/upload/strapi-server.ts
+++ b/example/src/extensions/upload/strapi-server.ts
@@ -1,6 +1,7 @@
 import imageOptimizerService from "strapi-plugin-image-optimizer/dist/server/services/image-optimizer-service";
+import { LoadedPlugin } from "@strapi/types/dist/types/core/plugins";
 
-module.exports = (plugin) => {
+module.exports = (plugin: LoadedPlugin) => {
   plugin.services["image-manipulation"] = imageOptimizerService;
   return plugin;
 };

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "strapi-plugin-image-optimizer",
-  "version": "2.1.1",
+  "version": "2.2.0",
   "description": "Optimize your images for desktop, tablet and mobile and different image formats.",
   "homepage": "https://github.com/marlokessler/strapi-plugin-image-optimizer#readme",
   "license": "MIT",
@@ -47,18 +47,19 @@
     ]
   },
   "engines": {
-    "node": ">=14.19.1 <=18.x.x",
+    "node": ">=18.0.0 <=20.x.x",
     "npm": ">=6.0.0"
   },
   "devDependencies": {
-    "@strapi/typescript-utils": "^4.11.7",
+    "@strapi/typescript-utils": "^4.20.5",
     "install-peers": "^1.0.4",
-    "typescript": "^5.0.4"
+    "typescript": "^5.4.2"
   },
   "peerDependencies": {
-    "@strapi/strapi": "^4.11.7",
-    "@strapi/utils": "^4.11.7",
-    "yup": "^1.1.1"
+    "@strapi/core": "*",
+    "@strapi/strapi": "^4.20.5",
+    "@strapi/utils": "^4.20.5",
+    "yup": "^1.4.0"
   },
   "scripts": {
     "add:apple_arm": "npm_config_build_from_source=true yarn add",

--- a/server/services/image-optimizer-service.ts
+++ b/server/services/image-optimizer-service.ts
@@ -1,9 +1,11 @@
 import { ReadStream, createReadStream, createWriteStream } from "fs";
 import { join } from "path";
 import sharp, { Sharp, Metadata } from "sharp";
+import { file as fileUtils } from '@strapi/utils';
 
-import { bytesToKbytes } from "@strapi/utils/dist/file";
-import imageManipulation from "@strapi/plugin-upload/server/services/image-manipulation";
+// @ts-ignore - No types available
+import pluginUpload from "@strapi/plugin-upload/strapi-server";
+const imageManipulation = pluginUpload().services["image-manipulation"];
 
 import {
   OutputFormat,
@@ -113,7 +115,7 @@ async function resizeFileTo(
     path: sourceFile.path,
     width: metadata.width,
     height: metadata.height,
-    size: metadata.size && bytesToKbytes(metadata.size),
+    size: metadata.size && fileUtils.bytesToKbytes(metadata.size),
     getStream: () => createReadStream(filePath),
   };
 }

--- a/server/services/settings-service.ts
+++ b/server/services/settings-service.ts
@@ -1,5 +1,5 @@
 import {} from "@strapi/strapi";
-import { Config } from "../models/config";
+import { Config } from "../models";
 import pluginId from "../utils/pluginId";
 
 class SettingsService {

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -3,7 +3,7 @@
   "compilerOptions": {
     "outDir": "dist",
     "rootDir": ".",
-    "declaration": true
+    "declaration": true,
   },
   "include": [
     // Include the root directory

